components/script/layout_image.rs

```diff
@@ -39,8 +47,22 @@ impl FetchResponseListener for LayoutImageContext {
     }
 
     fn process_response_eof(&mut self, response: Result<(), NetworkError>) {
+        // TODO notify_pending_response doesn't use the ResourceFetchTiming
         self.cache.notify_pending_response(self.id,
-                                           FetchResponseMsg::ProcessResponseEOF(response));
+                                           FetchResponseMsg::ProcessResponseEOF(
+                                           response.map(|_| ResourceFetchTiming::new())));
+    }
+
+    fn resource_timing(&mut self) -> &mut ResourceFetchTiming {
+        &mut self.resource_timing
+    }
+
+    fn submit_resource_timing(&self) {
+        let local_name = DOMString::from("other");
+        let global = self.doc.root().global();
+        let entry = PerformanceResourceTiming::new(
+            &global, global.get_url().clone(), local_name, None, &self.resource_timing);
```

Comment:
We should store the url from fetch_image_for_layout in LayoutImageContext instead of using the global's URL here.